key-manager-oapi.yaml

```diff
@@ -0,0 +1,223 @@
+                type: object
+                properties:
+                  statuses:
+                    type: array
+                    description: Status result of each `request.keystores` with same length and order of `request.keystores`
+                      type: object
+                      properties:
+                        status:
+                          type: string
+                          description: |
+                            - imported: Keystore successfully decrypted and imported to keymanager permanent storage
+                            - duplicate: Keystore's pubkey is already known to the keymanager
+                            - error: Any other status different to the above: decrypting error, I/O errors, etc.
+                          enum:
+                            - imported
+                            - duplicate
+                            - error
+                        message:
+                          type: string
+                          description: error message if status == error
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalError"
+
+    delete:
+      operationId: DeleteKeys
+      summary: Delete Keys.
+      description: |
+        DELETE must delete all keys from `request.pubkeys` that are known to the keymanager and exist in its
+        persistent storage. Additionally, DELETE must fetch the slashing protection data for the requested keys from
+        persistent storage, which must be retained (and not deleted) after the response has been sent. Therefore in the
+        case of two identical delete requests being made, both will have access to slashing protection data.
+
+        In a single atomic sequential operation the keymanager must:
+        1. Guarantee that key(s) can not produce any more signature; only then
+        2. Delete key(s) and serialize its associated slashing protection data
+
+        DELETE should never return a 404 response, even if all pubkeys from request.pubkeys have no extant keystores
+        nor slashing protection data.
+      security:
+        - cookieAuth: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                pubkeys:
+                  type: array
+                  description: List of public keys to delete.
+                  items:
+                    $ref: "#/components/types/Pubkey"
+      responses:
+        "200":
+          description: Success response
+          content:
+            application/json:
+              schema:
+                title: DeleteKeysResponse
+                type: object
+                properties:
+                  statuses:
+                    type: array
+                    description: List of public keys successfully deleted.
```

Comment:
The list contains all keys with their statuses, not only the ones successfully deleted.